smarts/core/smarts.py

```diff
@@ -108,7 +108,7 @@ def __init__(
         self._reset_required = False
 
         assert fixed_timestep_sec is None or fixed_timestep_sec > 0
-        self.fixed_timestep_sec: float = fixed_timestep_sec
+        self.fixed_timestep_sec: Optional[float] = fixed_timestep_sec
```

Comment:
Why use `Optional` here?

`Optional[...]` is a shorthand notation for `Union[..., None]` .

```suggestion
        self.fixed_timestep_sec: float = fixed_timestep_sec
```